include/zephyr/kernel.h

```diff
@@ -5443,7 +5443,9 @@ struct k_poll_event {
 
 #define K_POLL_EVENT_INITIALIZER(_event_type, _event_mode, _event_obj) \
 	{ \
+	._node = {}, \
```

Comment:
This is a dlist node with subfields, isn't it then subject to the same warning?

Reply:
The compiler in question: arm-none-eabi-g++ (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]

The very helpful error message when calling K_POLL_EVENT_STATIC_INITIALIZER from inside another macro in a C++ file:  sorry, unimplemented: non-trivial designated initializers not supported

There are probably a few other environmental/Kconfig settings that lead to this admittedly obscure and unique problem. And to be fair using a different cross compiler on RISC-V platform did NOT have this same issue.  My motivation for pushing these changes are complete selfish, I'd like to minimize the set of patches we have to maintain across versions. So I don't disagree with your assertions that this PR is incomplete (sisyphean). For our use case C++ represents a small subset of our codebase that interacts with Zephyr primitives.

Regarding your question about the dlist node: I believe the issue is with using designated initializers, which in C++ seems to enforce field order and completeness (except trailing fields) with varying degrees of nitpickyness depending on compiler version and flags.